statistics-rest/src/main/java/org/uniprot/api/statistics/service/StatisticService.java

```diff
@@ -0,0 +1,85 @@
+        this.statisticMapper = statisticMapper;
+    }
+
+    public Collection<StatisticModuleStatisticCategory> findAllByVersionAndStatisticTypeAndCategoryIn(
+            String version, String statisticType, Collection<String> categories) {
+        List<UniprotkbStatisticsEntry> entries;
+        if (categories.isEmpty()) {
+            entries =
+                    statisticsEntryRepository.findAllByReleaseNameAndEntryType(
+                            version, statisticMapper.map(getStatisticType(statisticType)));
+        } else {
+            entries =
+                    statisticsEntryRepository
+                            .findAllByReleaseNameAndEntryTypeAndStatisticsCategoryIn(
+                                    version,
+                                    statisticMapper.map(getStatisticType(statisticType)),
+                                    categories.stream()
+                                            .map(
+                                                    cat ->
+                                                            statisticsCategoryRepository
+                                                                    .findByCategoryIgnoreCase(cat)
+                                                                    .orElseThrow(
+                                                                            () ->
+                                                                                    new IllegalArgumentException(
+                                                                                            String
+                                                                                                    .format(
+                                                                                                            "Invalid Statistic Category: %s",
+                                                                                                            cat))))
+                                            .collect(Collectors.toList()));
+        }
+        return entries.stream()
+                .collect(Collectors.groupingBy(UniprotkbStatisticsEntry::getStatisticsCategory))
+                .entrySet()
+                .stream()
+                .map(entry -> StatisticModuleStatisticCategoryImpl.builder()
+                        .categoryName(entry.getKey().getCategory())
+                        .totalCount(
+                                entry.getValue().stream().mapToLong(UniprotkbStatisticsEntry::getValueCount).sum())
+                        .label(entry.getKey().getLabel())
+                        .searchField(entry.getKey().getSearchField())
+                        .items(
+                                entry.getValue().stream()
+                                        .map(statisticMapper::map)
+                                        .collect(Collectors.toList()))
+                        .build())
+                .collect(Collectors.toList());
```

Comment:
this code block is hard to read. Please split it into multiple statements.
take out the code in map entry-> StatisticModuleStatisticCategoryImpl to a method and use method reference.

Reply:
done